src/PJ_horner.c

```diff
@@ -393,434 +408,117 @@ static int horner_silence (int i) {
-    }
-    else { /* collect output coord,switch to reverse checking */
-      *Nout = N;
-      *Eout = E;
-      Cp    = C_r;
-    }
-
-  /* tol-check of results*/
-  if (res == 0 && (fabs(N - N_in) < tol && fabs(E - E_in) < tol))
-      return (0);
-  else if (res == 0) res = TRF_AREA_;
-  return(res);
-
-#undef   FLOATLIMIT
-
+static PJ_OBS horner_reverse_obs (PJ_OBS point, PJ *P) {
+    point.coo.uv = horner ((HORNER *) P->opaque, -1, point.coo.uv);
+    return point;
 }
 
 
-/* s45b polynomia */
-  static  double      C_ttu_b[] = {
-    /* tc32_ed50 -> utm32_ed50 : Bornholm */
-    /* m_lim_gen:  0.086  red  = 0      OBS   =    852   */
-    /* m = 1.38 cm my_loss =    +2  y_enp = +10.5 */
-    /* m = 1.44 cm mx_loss =    +2  x_enp = +10.4 */
-
-    /*deg*/  4.0,
-    /*Poly NORTH :: e-degree =  0  : n-degree =  4 */
-    /*  0*/   6.1258112678e+06,   9.9999971567e-01,   1.5372750011e-10,
-    /*  3*/   5.9300860915e-15,   2.2609497633e-19,
-
-    /*Poly NORTH :: e-degree =  1  : n-degree =  3 */
-    /*  5*/   4.3188227445e-05,   2.8225130416e-10,   7.8740007114e-16,
-    /*  8*/  -1.7453997279e-19,
-
-    /*Poly NORTH :: e-degree =  2  : n-degree =  2 */
-    /*  9*/   1.6877465415e-10,  -1.1234649773e-14,  -1.7042333358e-18,
-    /*Poly NORTH :: e-degree =  3  : n-degree =  1 */
-    /* 12*/  -7.9303467953e-15,  -5.2906832535e-19,
-    /*Poly NORTH :: e-degree =  4  : n-degree =  0 */
-    /* 14*/   3.9984284847e-19,
-
-    /*tcy*/    6125810.306769,
-
-    /*Poly EAST  :: n-degree =  0  : e-degree =  4 */
-    /*  0*/   8.7760574982e+05,   9.9999752475e-01,   2.8817299305e-10,
-    /*  3*/   5.5641310680e-15,  -1.5544700949e-18,
-
-    /*Poly EAST  :: n-degree =  1  : e-degree =  3 */
-    /*  5*/  -4.1357045890e-05,   4.2106213519e-11,   2.8525551629e-14,
-    /*  8*/  -1.9107771273e-18,
-
-    /*Poly EAST  :: n-degree =  2  : e-degree =  2 */
-    /*  9*/   3.3615590093e-10,   2.4380247154e-14,  -2.0241230315e-18,
-    /*Poly EAST  :: n-degree =  3  : e-degree =  1 */
-    /* 12*/   1.2429019719e-15,   5.3886155968e-19,
-    /*Poly EAST  :: n-degree =  4  : e-degree =  0 */
-    /* 14*/  -1.0167505000e-18,
 
-    /*tcx*/     877605.269066
-  };
 
-  static  double      C_utt_b[] = {
-    /* utm32_ed50 -> tc32_ed50 : Bornholm */
-    /* m_lim_gen:  0.086  red  = 0      OBS   =    852   */
-    /* m = 1.38 cm my_loss =    +2  y_enp = +10.8 */
-    /* m = 1.44 cm mx_loss =    +2  x_enp = +10.7 */
-
-    /*deg*/  4.0,
-    /*Poly NORTH :: e-degree =  0  : n-degree =  4 */
-    /*  0*/   6.1258103208e+06,   1.0000002826e+00,  -1.5372762184e-10,
-    /*  3*/  -5.9304261011e-15,  -2.2612705361e-19,
-
-    /*Poly NORTH :: e-degree =  1  : n-degree =  3 */
-    /*  5*/  -4.3188331419e-05,  -2.8225549995e-10,  -7.8529116371e-16,
-    /*  8*/   1.7476576773e-19,
+/**********************************************************************/
+static UV complex_horner (const HORNER *transformation, int direction, UV position) {
```

Comment:
This would be true for C++, but not for C where, syntactically, enums are ints. I do, however, agree, that the argument should be of type enum pj_direction. It was not, originally, since I had tried to keep as close as possible to the Poder/Engsager code. Now, that I have tightened it up significantly anyway, I might as well implement this also.